crates/hotshot/src/block_impl.rs

```diff
@@ -0,0 +1,174 @@
+//! This module provides an implementation of the `HotShot` suite of traits.
+use std::{
+    collections::HashSet,
+    fmt::{Debug, Display},
+    ops::Deref,
+};
+
+use commit::{Commitment, Committable};
+use hotshot_types::traits::{block_contents::Transaction, state::TestableBlock, BlockPayload};
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+
+/// The transaction in a [`VIDBlockPayload`].
+#[derive(PartialEq, Eq, Hash, Serialize, Deserialize, Clone, Debug)]
+pub struct VIDTransaction {
+    /// identifier for the transaction
+    pub id: u64,
+    /// padding to add to txn (to make it larger and thereby more realistic)
+    pub padding: Vec<u8>,
```

Comment:
If this is going to be a more generic implementation that is used outside the demo, in real applications, we shouldn't impose any structure beyond just bytes, ie `struct VIDTransaction(Vec<u8>)`

Reply:
I'll double-check how `id` is used since I think we might still need it for logging/testing purposes. If so, maybe I could add a feature flag.

Reply:
Ideally we would build the demo application (which uses `id`?) on top of the VID primitive in the same way we will build the sequencer application on top of it: have a demo-specific transaction type with internal structure, but serialize/deserialize when sending it to/receiving it from HotShot APIs

Reply:
It turned out that `id` wasn't used in many places so I just removed it. Agree that we can have a demo-specific transaction type if needed.